gammapy/maps/core.py

```diff
@@ -981,6 +981,61 @@ def _resample_by_idx(self, idx, weights=None, preserve_counts=False):
         """
         pass
 
+    def reproject_spatial_geom(self, geom, preserve_counts=False, oversampling_factor=10):
+        """Reproject map to input geometry using oversampling :
+           the base map is oversampled and its bins are re-distributed
+           into a map with the target geom.
+           Note that alternative methods can be found in the reproject package.
+        
+        Parameters
+        ----------
+        geom : `~gammapy.maps.Geom`
+            2D Map geometry
+        preserve_counts : bool
+            Preserve the integral over each bin.  This should be true
+            if the map is an integral quantity (e.g. counts) and false if
+            the map is a differential quantity (e.g. intensity)
+        oversampling_factor : int
+           Minimal factor between the bin size of the output map and the oversampled base map.
+           Used only for the oversampling method.
+
+        Returns
+        -------
+        output_map : `Map`
+            Reprojected Map
+        """
+        from .region import RegionGeom
+        from .hpx import HpxGeom
+
+        if isinstance(geom, RegionGeom) or isinstance(self.geom, RegionGeom):
```

Comment:
This should just work, why forbid it? I think it is basically equivalent to `.to_region_nd_map()`, but still. Or when raising an error, mention `.to_region_nd_map()` as an alternative solution.

Reply:
`pixel_scales` is not defined

Reply:
I see, one would need to work with `RegionGeom.to_wcs_geom().pixel_scales`...